sapientml_core/templates/model_templates/model.py.jinja

```diff
@@ -22,27 +22,45 @@ model = {{ model_name }}({{ silent }}random_state=random_state_model, {{ params
 from sklearn.multioutput import MultiOutputRegressor
 
 model = MultiOutputRegressor(model)
+
 {% elif is_multioutput_classification %}
 from sklearn.multioutput import MultiOutputClassifier
 
 model = MultiOutputClassifier(model)
 {% endif %}
+
 {% set xgbclassifier = "XGBClassifier" %}
-{% if model_name == xgbclassifier %}
-from sklearn.preprocessing import LabelEncoder
 
-label_encoder = LabelEncoder()
-target_train = pd.DataFrame(label_encoder.fit_transform(target_train), columns=TARGET_COLUMNS)
+{% if pipeline.is_multi_class_multi_targets %}
+from sklearn.preprocessing import LabelEncoder
+label_encoders = {}
+for i, column in enumerate(target_train.columns):
+    le = LabelEncoder()
+    target_train[column] = le.fit_transform(target_train[column])
+    label_encoders[column] = le
+{% elif model_name == xgbclassifier %}
+from sklearn.preprocessing import LabelEncoder
+label_encoders = LabelEncoder()
+target_train = pd.DataFrame(label_encoders.fit_transform(target_train), columns=TARGET_COLUMNS)
 {% endif %}
+
 {% if pipeline.task.target_columns|length == 1 %}
 model.fit(feature_train, target_train.values.ravel())
 {% else %}
 model.fit(feature_train, target_train)
 {% endif %}
+
 y_pred = model.predict(feature_test)
+
 {% if flag_predict_proba and (not pipeline.adaptation_metric.startswith("MAP_")) and (not pipeline.adaptation_metric == "LogLoss") and (pipeline.adaptation_metric not in metric_needing_predict_proba) %}
 y_pred = model.classes_[np.argmax(y_pred, axis=1)].reshape(-1, 1)
 {% endif %}
-{% if model_name == xgbclassifier and (not pipeline.adaptation_metric.startswith("MAP_")) and (not pipeline.adaptation_metric == "LogLoss") and (pipeline.adaptation_metric not in metric_needing_predict_proba) %}
-y_pred = label_encoder.inverse_transform(y_pred).reshape(-1, 1)
-{% endif %}
\ No newline at end of file
+
+{% if pipeline.is_multi_class_multi_targets %}
+y_pred_df = pd.DataFrame(y_pred, columns=TARGET_COLUMNS)
+for column in TARGET_COLUMNS:
+    y_pred_df[column] = label_encoders[column].inverse_transform(y_pred_df[column].astype(int))
+y_pred = y_pred_df
```

Comment:
This line defines y_pred as `pandas.DataFrame`, but I've found that this causes an error in `disp = ConfusionMatrixDisplay.from_predictions(target_test[column], y_pred[:, i])` in templates/other_templates/confusion_matrix.py.jinja, which assumes y_pred is a numpy array. Could you modify the issue?

Reply:
Thanks for catching it. Fix in [7c40063](https://github.com/sapientml/core/pull/59/commits/7c400633c01f1df35a9fd61df4c74a1280e28e73)

Reply:
LGTM.